### ISSUE TITLE
Crash on likely non-compliant server

### ISSUE THREAD

**Issue by lnicola** [stac-utils/pystac#1178]
I noticed that `stac-client collections https://datahub.creodias.eu/stac/` crashes in a pretty obscure way (a NoneError, more or less).

The server is probably not exactly compliant, but it would be nice to report a more useful error.

**Comment by jsignell** [comment]
If you check the collections at https://datahub.creodias.eu/stac/collections you will see that the spatial extents are indeed set to null.

**Comment by jsignell** [comment]
Right. I think the most pystac-client will do is try to provide a more informative error and rely on the STAC provider to provide a compliant STAC catalog.
